lib/index.js

```diff
@@ -536,11 +535,22 @@ api.verify = async ({
     });
   }
 
+  let combinedHash;
+  if(clientId) {
+    combinedHash = prefixedHash(`${hash}:${clientId}`);
+  } else {
+    combinedHash = prefixedHash(hash);
+  }
+
+  let nonceVerified;
   let tokenValue = record.meta['bedrock-authn-token'].tokens[type];
   const now = new Date();
   if(type === 'nonce') {
     for(const token of tokenValue) {
-      if(token.sha256 === prefixedHash(hash)) {
+      nonceVerified = crypto.timingSafeEqual(
+        new Buffer(token.sha256, 'base64'),
+        new Buffer(combinedHash, 'base64'));
+      if(nonceVerified) {
```

Comment:
There's still a bug in this section where -- if the the `combinedHash` doesn't match *any* of the tokens that the `tokenValue` will be set to the array of tokens instead of being undefined, right? It seems like maybe we can return `false` early when that happens? Can you check on that?

Reply:
I think it might be better to throw an error here rather than return `false` early. What do you think?

Reply:
Well it looks like the test is set up to return `false` if a different hash is passed. So I will keep it that way.